tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/jdbc/RelationalDataSource.java

```diff
@@ -18,7 +18,7 @@
 import java.util.Iterator;
 import java.util.List;
 
-public interface JdbcTableDataSource
+public interface RelationalDataSource
```

Comment:
There's a difference between `HiveTableDefinition` and `RelationalTableDefinitions` because there are some Hadoop specific things which we want to express there, e.g. file type, compression. I tried to rewrite that but it was PITA because `HiveTableDefitnition` heavily depends on the fact that data source is of type `HiveDataSource`. And even if it's not a `HiveTableDefinition` depending on that directly it's exposed further through `getDataSource()` method and it's used in e.g. `HiveTableManager`. This dependency simply leaks further...

Anyway, it's way beyond the scope of this PR. For now Cassandra and JDBC databases uses exactly the same properties to express tables so I don't see the reason to not reuse that code. When we notice significant differences between `JDBCTableDefinition` and `CassandraTableDefinition` we can rename back `RelationalTableDefinition` to `JDBCTableDefinition` and introduce `CassandraTableDefinition`. For now I believe we should reuse code which would be identical anyway...